### PR TITLE
core: Updated docstring for RunnablePick

### PR DESCRIPTION
**Pull request by devesh-2002**
**Description:** : Updated the docstring for RunnablePick. Added Overview and an Example for RunnablePick class.
   **Issue:** : #18803

### CODE REVIEW

File: libs/core/langchain_core/runnables/passthrough.py [hunk 1]
```diff
@@ -611,6 +611,26 @@ async def input_aiter() -> AsyncIterator[Dict[str, Any]]:
 class RunnablePick(RunnableSerializable[Dict[str, Any], Dict[str, Any]]):
     """
     Runnable that picks keys from Dict[str, Any] inputs.
+
+    `RunnablePick` class represents a runnable that selectively picks keys from a dictionary input. It allows you to specify one or more keys to extract from the input dictionary and returns a new dictionary containing only the selected keys.
+
+    Example :
```
Comment (hwchase17) [inline libs/core/langchain_core/runnables/passthrough.py:617]: i think the indentation may be off here

**Comment by devesh-2002** [comment]
Can you check the indentation now? @hwchase17

File: libs/core/langchain_core/runnables/passthrough.py [hunk 2]
```diff
@@ -611,6 +611,28 @@ async def input_aiter() -> AsyncIterator[Dict[str, Any]]:
 class RunnablePick(RunnableSerializable[Dict[str, Any], Dict[str, Any]]):
     """
     Runnable that picks keys from Dict[str, Any] inputs.
+
+    RunnablePick class represents a runnable that selectively picks keys from a dictionary input. 
+    It allows you to specify one or more keys to extract from the input dictionary.
+    It returns a new dictionary containing only the selected keys.
+
+    Example : 
+    .. code-block:: python
```
Comment (eyurtsev) [inline libs/core/langchain_core/runnables/passthrough.py:620]: ..code-block and the contents needs to be further indented with additional tab -- you can use the other code-block examples as reference

**Comment by devesh-2002** [comment]
Is the indentation correct now? @eyurtsev

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
Awesome thank you!

**Pull request merged**